docs/how_to/composability.md

```diff
@@ -0,0 +1,23 @@
+# Composability
+
+
+GPT Index offers **composability** of your indices, meaning that you can build indices on top of other indices. This allows you to more effectively index your entire document tree in order to feed custom knowledge to GPT.
+
+Composability allows you to to define lower-level indices for each document, and higher-order indices over a collection of documents. To see how this works, imagine defining 1) a tree index for the text within each document, and 2) a list index over each tree index (one document) within your collection.
```

Comment:
I'm wondering if we can unpack this line a little by splitting it up + introducing more diagrams + code snippets. Here's an example:

To see how this works, imagine you have 3 documents: `doc1`, `doc2`, and `doc3`

```python
doc1 = SimpleDirectoryReader('data1').load_data()
doc2 = SimpleDirectoryReader('data2').load_data()
doc3 = SimpleDirectoryReader('data3').load_data()
```
<img width="498" alt="image" src="https://user-images.githubusercontent.com/10181537/207229970-f5e3807e-28ab-4691-85d7-fc310604b61d.png">
(something like the above but with no arrows)

Now let's define a tree index for each document. In python, we have:
```python
index1 = GPTTreeIndex(doc1)
index2 = GPTTreeIndex(doc2)
index2 = GPTTreeIndex(doc2)
```
Visually, we have:
<img width="528" alt="image" src="https://user-images.githubusercontent.com/10181537/207230272-13245414-345e-4835-9f1f-625797cfcd37.png">
(like this but no arrows)

We can create a list index on these 3 tree indices:

```python
list_index = GPTListIndex([index1, index2, index3])
```
(👇 then your original takes it from here!)

Reply:
these are _great_ suggestions! thanks so much